Cargo.toml

```diff
@@ -22,6 +22,9 @@ byteorder = { version = "1.3.1", default-features = false }
 failure = { version = "0.1.5", default-features = false, features = ["derive"] }
 subtle = { version = "2.0.0", default-features = false }
 libm = "0.1.4"
+signature = { version = "0.2.0", default-features = false }
+sha2 = { version = "0.8", default-features = false }
+sha-1 = { version = "0.8.1", default-features = false }
```

Comment:
General note, I would really like to not have hard dependencies on hashing crates. Not sure if it can be avoided.